### PR TITLE
make use of #tsqueryreq.allow_qbuf_reuse

### PR DESCRIPTION
**Pull request by hmmr**
RTS-733, RTS-870, RTS-1006

Depends on https://github.com/basho/riak_pb/pull/208, required by https://github.com/basho/riak_test/pull/1152. Part of https://github.com/basho/riak_kv/pull/1479 and related PRs.

Extract and incorporate the `allow_qbuf_reuse` property in `#tsqueryreq`. This is used to control query buffer reuse for ORDER BY queries.

### CODE REVIEW

**Comment by gordonguthrie** [comment]
+1 1b030a7

**Comment by javajolt** [comment]
:+1: 9d3909b

**Comment by hmmr** [comment]
@borshop merge

File: rebar.config [hunk 1]
```diff
@@ -13,7 +13,7 @@
 ]}.
 
 {deps, [
-    {riak_pb, ".*", {git, "https://github.com/basho/riak_pb", {tag, "2.2.0.0"}}}
+    {riak_pb, ".*", {git, "git://github.com/basho/riak_pb", {tag, "2.2.0.1"}}}
```
Comment (javajolt) [inline rebar.config]: This should be `https`, the new standard

Comment (hmmr) [inline rebar.config]: fixed in 9a3dbe9

**Comment by javajolt** [comment]
:+1: 9a3dbe9

**Comment by javajolt** [comment]
@borshop merge

**Pull request merged**